projects/vdk-plugins/vdk-trino/tests/test_vdk_trino_lineage.py

```diff
@@ -26,27 +28,225 @@
+            context.state.set(LINEAGE_LOGGER_KEY, mock_lineage_logger)
+
+    runner = CliEntryBasedTestRunner(TestConfigPlugin(), trino_plugin)
+
+    result: Result = runner.invoke(
+        ["trino-query", "--query", f"create table {table_name} (test_column varchar)"]
+    )
+    cli_assert_equal(0, result)
+
+    select_query = f"select * from {table_name}"
+    result: Result = runner.invoke(["trino-query", "--query", select_query])
+    cli_assert_equal(0, result)
+
+    # the lineage data is different on every run of the test,
+    # so we need this class to generalize the dict which is to be matched
+    class SelectLineageDataMatch:
+        def __eq__(self, lineage_data):
+            import json
+
+            print("SELECT query lineage data:")
+            print(json.dumps(lineage_data, indent=4))
+            return (
+                lineage_data.keys()
+                >= {"@type", "query", "@id", "status", "inputTableColumnInfos"}
+                and lineage_data["@type"] == "select"
+                and lineage_data["status"] == "OK"
+                and lineage_data["query"] == select_query
+                and re.fullmatch(r"[0-9]{10}\.[0-9]+", lineage_data["@id"])
+                and lineage_data["inputTableColumnInfos"][0]["table"]["schemaTable"][
+                    "table"
+                ]
+                == table_name
+            )
+
+    mock_lineage_logger.send.assert_called_with(SelectLineageDataMatch())
+
+
+@pytest.mark.usefixtures("trino_service")
+@mock.patch.dict(
+    os.environ,
+    {
+        VDK_DB_DEFAULT_TYPE: "TRINO",
+        VDK_TRINO_PORT: "8080",
+        VDK_TRINO_USE_SSL: "False",
+    },
+)
+def test_lineage_for_insert_select():
+    table_name_from = "test_table_" + uuid.uuid4().hex
+    table_name_to = "test_table_" + uuid.uuid4().hex
+
+    mock_lineage_logger = mock.MagicMock(LineageLogger)
 
     class TestConfigPlugin:
         @hookimpl
         def vdk_initialize(self, context):
-            context.state.set(LINEAGE_LOGGER_KEY, mock_lineage)
+            context.state.set(LINEAGE_LOGGER_KEY, mock_lineage_logger)
 
     runner = CliEntryBasedTestRunner(TestConfigPlugin(), trino_plugin)
 
-    result: Result = runner.invoke(["trino-query", "--query", "SELECT 1"])
+    result: Result = runner.invoke(
+        [
+            "trino-query",
+            "--query",
+            f"create table {table_name_from} (test_column varchar)",
+        ]
+    )
+    cli_assert_equal(0, result)
+
+    result: Result = runner.invoke(
+        [
+            "trino-query",
+            "--query",
+            f"create table {table_name_to} (test_column varchar)",
+        ]
+    )
+    cli_assert_equal(0, result)
 
-    class LineageDataMatch:  # the lineage data is different on every run of the test,
-        # so we need this class to generalize the dict which is to be matched
+    insert_select_query = f"insert into {table_name_to} select * from {table_name_from}"
```

Comment:
Maybe add tests that include fully qualified names (catalog.schema.table) and (schema.table) 

?

Reply:
I am handling those in my next change. Brace yourself, it's a big change :D